docs/docs/apis/providers/supplier-interfaces-api.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Supplier Interfaces API"
+description: "Extensible interfaces for custom backend data providers"
+sidebar_position: 2
+---
+
+# Supplier Interfaces API
+
+The Supplier Interfaces API provides extensible interfaces for integrating custom data providers with the Cardano Client Library. These interfaces enable custom backend implementations, service extensions, and flexible provider switching for enhanced testability and integration capabilities.
+
+## Key Features
+
+- **Extensible Architecture**: Define custom data provider implementations
+- **Interface Standardization**: Consistent contracts for different data sources
+- **Provider Flexibility**: Switch between implementations seamlessly
+- **Testing Support**: Mock implementations for unit and integration testing
+- **Service Composition**: Combine multiple suppliers for complex scenarios
+- **Custom Integration**: Integrate with proprietary or specialized data sources
+
+## Core Classes
+
+### Supplier Interfaces
+- `UtxoSupplier` - Provides UTXO data for addresses and transactions
+- `ProtocolParamsSupplier` - Supplies current protocol parameters
+- `TransactionProcessor` - Handles transaction submission and evaluation
+- `ScriptSupplier` - Provides Plutus scripts by hash
+- `TransactionEvaluator` - Evaluates transaction execution costs
+
+### Built-in Implementations
+- `OgmiosProtocolParamSupplier` - Ogmios-based protocol parameters
+- `OgmiosTransactionProcessor` - Ogmios transaction processing
+- `KupoUtxoSupplier` - Kupo-based UTXO provider
+- `OgmiosTransactionEvaluator` - Ogmios script evaluation
+
```

Comment:
Please mention `DefaultUtxoSupplier`, `DefaultProtocolParamSupplier`, and `DefaultTransactionProcessor`, which are the default implementations instantiated from `BackendService` and are currently set as the defaults for transaction building."